### PR TITLE
feat: (TNLT-6397) Article read animation

### PR DESCRIPTION
**Pull request by aronshamash**
- https://nidigitalsolutions.jira.com/browse/TNLT-6397
- For articles that have been deemed as being read we need to differentiate the article tiles that have already be shown to the user (`highlight:false`) and recently read articles which have not yet been shown (`highlight:true`). 
- The `highlight` flag allows TCN to distinguish which article tiles should fade out to reduced opacity and which should have reduced opacity initially without animation.
- `Tile AH` used by the Cartoon slice needed some refactoring because the tile was rendering the byline using the `Byline` component rather than `ArticleSummary` like the rest of the other tiles.
- `isOpinionByline` flag is not used so it was removed from `ArticleSummary`

Agreed data format:

```
    const readArticles = [
      {
        id: "de2749dc-6d4c-11eb-bc24-b1bb9cd3bc5f",
        highlight: true,
      },
      {
        id: "29f1f8e0-6d4c-11eb-bc24-b1bb9cd3bc5f",
        highlight: false,
      },
    ];
```

### iOS
https://user-images.githubusercontent.com/1736782/108087978-dd374600-706f-11eb-8e4d-0b4bcfba3c44.mov

### Android

https://user-images.githubusercontent.com/1736782/108191323-e4f7f880-710a-11eb-8286-a4a9c73d5aed.mov

### CODE REVIEW

File: packages/pages/src/section/section.js [hunk 1]
```diff
@@ -142,7 +142,24 @@ class SectionPage extends Component {
       sectionName: section.name,
     });
 
-    const readArticles = [];
+    const readArticles = [
```
Comment (aronshamash) [inline packages/pages/src/section/section.js:145]: remove before merging

File: packages/edition-slices/src/tiles/shared/tile-summary.tsx [hunk 2]
```diff
@@ -64,158 +56,37 @@ const TileSummary: React.FC<Props> = ({
...
-                  markAsRead,
-                }}
-                strapline={renderStrapline(markAsRead)}
-                saveStar={withStar && renderSaveStar()}
-                style={style}
-              />
-            );
-          }}
+          {({ readArticles }) => (
+            <ArticleSummary
```
Comment (aronshamash) [inline packages/edition-slices/src/tiles/shared/tile-summary.tsx:64]: Now have a component in order to pass in `readArticles` as a prop so don't need to worry about managing state.

File: packages/article-summary/src/article-summary.js [hunk 3]
```diff
@@ -86,13 +98,26 @@ function ArticleSummary(props) {
...
     <View style={style}>
-      {labelProps ? <ArticleSummaryLabel {...labelProps} /> : null}
-      {isOpinionByline && byline}
+      {labelProps ? (
+        <ArticleSummaryLabel
+          {...labelProps}
+          articleReadState={articleReadState}
+        />
+      ) : null}
+      {bylineOnTop && byline}
```
Comment (kamiranoff) [inline packages/article-summary/src/article-summary.js:120]: I think you have the opposite behaviour in line 131?
```
{!bylineOnTop && byline}
```

Comment (kamiranoff) [reply packages/article-summary/src/article-summary.js:120]: Oh. Ok. I understand.

File: packages/article-summary/src/article-summary.js [hunk 4]
```diff
@@ -21,18 +20,42 @@ function renderAst(ast) {
...
       delay: ARTICLE_READ_ANIMATION.delay,
       duration: ARTICLE_READ_ANIMATION.duration,
-      toValue: 0.6,
+      toValue: opacity,
       useNativeDriver: true,
     }).start();
-  }, [markAsRead]);
+  }, [articleReadState]);
+
+  return articleReadState.animate ? (
```
Comment (kamiranoff) [inline packages/article-summary/src/article-summary.js:36]: Minor but might be clearer to have two if/ switch statement rather than a nested ternary.

**Review by kamiranoff — APPROVED** [review]

**Review by c-j-j — APPROVED** [review]

File: packages/article-summary/src/article-summary.js (hunk 4 above)
Comment (aronshamash) [reply packages/article-summary/src/article-summary.js:36]: I think I'll leave as is this time. It seems to read ok to other devs.

File: packages/pages/src/section/section.js (hunk 1 above)
Comment (aronshamash) [reply packages/pages/src/section/section.js]: done

**Pull request merged**